File: app/src/main/java/org/xdty/callerinfo/di/PhoneStatusComponent.java
package org.xdty.callerinfo.di;

import org.xdty.callerinfo.di.modules.AppModule;
import org.xdty.callerinfo.di.modules.PhoneStatusModule;
import org.xdty.callerinfo.receiver.IncomingCall;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = { PhoneStatusModule.class, AppModule.class })
public interface PhoneStatusComponent {

    void inject(IncomingCall.IncomingCallListener listener);

}

File: app/src/main/java/org/xdty/callerinfo/di/PhoneStatusComponent.kt
package org.xdty.callerinfo.di

import dagger.Component
import org.xdty.callerinfo.di.modules.AppModule
import org.xdty.callerinfo.di.modules.PhoneStatusModule
import org.xdty.callerinfo.receiver.IncomingCall.PhoneStateListener
import javax.inject.Singleton

@Singleton
@Component(modules = [PhoneStatusModule::class, AppModule::class])
interface PhoneStatusComponent {
    fun inject(phoneStateListener: PhoneStateListener)
}